packages/migration/migrations/hearth/20221110092434-add_identifiers_DOB_gender_location.js

```diff
@@ -0,0 +1,182 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+
+import {
+  COLLECTION_NAMES,
+  getCompositionCursor,
+  getCollectionDocuments,
+  getTotalDocCountByCollectionName
+} from '../../utils/hearth-helper.js'
+
+import {
+  updateComposition,
+  updateFieldNameByCompositionId
+} from '../../utils/elasticsearch-helper.js'
+
+export const up = async (db, client) => {
+  const session = client.startSession()
+  const limit = 10
+  let skip = 0
+  let processedDocCount = 0
+  try {
+    await session.withTransaction(async () => {
+      //rename field name declarationLocationHirarchyIds to declarationJurisdictionIds on elasticSearch
+      await updateFieldNameByCompositionId(
+        'declarationJurisdictionIds',
+        'declarationLocationHirarchyIds'
+      )
+      const totalCompositionCount = await getTotalDocCountByCollectionName(
+        db,
+        COLLECTION_NAMES.COMPOSITION
+      )
+
+      while (totalCompositionCount > processedDocCount) {
+        const compositionCursor = await getCompositionCursor(db, limit, skip)
+        const count = await compositionCursor.count()
```

Comment:
Will count be different than limit?

Reply:
Yeah, It might be when the limit is 50 but there are actually less than 50 documents in mongo.